factory/django.py

```diff
@@ -159,6 +182,102 @@ def _get_or_create(cls, model_class, *args, **kwargs):
 
         return instance
 
+    @classmethod
+    def supports_bulk_insert(cls):
+        return (cls._meta.use_bulk_create
+                and connection_supports_bulk_insert(cls._meta.database))
+
+    @classmethod
+    def create(cls, **kwargs):
+        """Create an instance of the associated class, with overridden attrs."""
+        if not cls.supports_bulk_insert():
+            return super().create(**kwargs)
+
+        return cls._bulk_create(1, **kwargs)[0]
+
+    @classmethod
+    def create_batch(cls, size, **kwargs):
+        if not cls.supports_bulk_insert():
+            return super().create_batch(size, **kwargs)
+
+        return cls._bulk_create(size, **kwargs)
+
+    @classmethod
+    def _refresh_database_pks(cls, model_cls, objs):
+        # Avoid causing a django.core.exceptions.AppRegistryNotReady throughout all the tests.
+        # TODO: remove the `from . import django` from the `__init__.py`
+        from django.contrib.contenttypes.fields import GenericForeignKey
+
+        def get_field_value(instance, field):
+            if isinstance(field, GenericForeignKey) and field.is_cached(instance):
+                return field.get_cached_value(instance)
+            return getattr(instance, field.name)
+
+        """
+        Current Django version's GenericForeignKey is not made to work with bulk_insert.
+
+        The issue is that it caches the object referenced, once the object is
+        saved and receives a pk, the cache no longer matches. It doesn't
+        matter that it's the same obj reference. This is to bypass that pk
+        check and reset it.
+        """
```

Comment:
We should use `#` for comments and multiline strings for docstrings.

Reply:
I thought this looked better. Done.